src/request.ts

```diff
@@ -44,23 +44,24 @@ export const buildHeaders = (
   etag: string | undefined,
   contentType: string | undefined,
   custom: CustomHeaders | undefined,
-): Headers => {
-  const head = new Headers();
+): Record<string, string> => {
+  const head: Record<string, string> = {};
   if (appName) {
-    head.append('UNLEASH-APPNAME', appName);
-    head.append('User-Agent', appName);
+    head['UNLEASH-APPNAME'] = appName;
+    head['User-Agent'] = appName;
   }
   if (instanceId) {
-    head.append('UNLEASH-INSTANCEID', instanceId);
+    head['UNLEASH-INSTANCEID'] = instanceId;
   }
   if (etag) {
-    head.append('If-None-Match', etag);
+    head['If-None-Match'] = etag;
   }
   if (contentType) {
-    head.append('Content-Type', contentType);
+    head['Content-Type'] = contentType;
   }
   if (custom) {
-    Object.keys(custom).forEach((k) => head.append(k, custom[k]));
+    // eslint-disable-next-line no-return-assign
+    Object.keys(custom).forEach((k) => (head[k] = custom[k]));
```

Comment:
```suggestion
    Object.assign(head, custom);
```

I think that'll work